builtin/providers/mssql/config.go

```diff
@@ -0,0 +1,93 @@
+package mssql
+
+import (
+	"database/sql"
+	"fmt"
+	"os"
+
+	_ "github.com/denisenkom/go-mssqldb" //MS SQL db
+	"github.com/hashicorp/go-multierror"
+)
+
+// Config - provider config
+type Config struct {
+	Host                   string
+	Port                   int
+	Encrypt                string
+	TrustServerCertificate bool
+	Certificate            string
+	Username               string
+	Password               string
+}
+
+// Client struct holding connection string
+type Client struct {
+	username string
+	connStr  string
+}
+
+//NewClient returns new client config
+func (c *Config) NewClient() (*Client, error) {
+	// Connection String
+	var connStr string
+	// We need to validate some parameters
+	var errs []error
+
+	err := c.ValidateEncrypt()
+	if err != nil {
+		errs = append(errs, err)
+	}
+
+	connStr = fmt.Sprintf("server=%s;user id=%s;password=%s;port=%d", c.Host, c.Username, c.Password, c.Port)
+
+	switch c.Encrypt {
+	case "true":
+		connStr += fmt.Sprintf(";encrypt=%s;TrustServerCertificate=%t", c.Encrypt, c.TrustServerCertificate)
+		// If we have to check certificate we need certificate file
+		if !c.TrustServerCertificate {
+			if c.Certificate == "" {
+				return nil, fmt.Errorf("Please provide full path to file that contains public key certificate of the CA that signed the SQL. You specified certificate = '%s'", c.Certificate)
+			}
+			if _, err = os.Stat(c.Certificate); os.IsNotExist(err) {
+				return nil, fmt.Errorf("%s MS SQL certificate file cannot be found", c.Certificate)
+			}
+			connStr += fmt.Sprintf(";certificate=%s", c.Certificate)
```

Comment:
We've been trying to normalize Terraform options that expect file paths to instead expect file _contents_ and users can always use the `${file()}` interpolation function to pull in a path.

In this case it looks like the upstream API expects a path, so I think we'd need to get upstream to support direct certificate contents to be able to support it.

In that case this is AOK - just mentioning here so you know the project standard generally. :+1: